hosting/src/hooks/useGenkitArticle.tsx

```diff
@@ -0,0 +1,99 @@
+import {generateArticle} from "@/genkit/article";
+import {TanamDocumentClient} from "@/models/TanamDocumentClient";
+import {UserNotification} from "@/models/UserNotification";
+import {firestore, storage} from "@/plugins/firebase";
+import {TanamPublishStatus} from "@functions/models/TanamDocument";
+import {ArticleSchema} from "@functions/schemas/article";
+import {collection, doc, getDocs, limit, orderBy, query, setDoc, where} from "firebase/firestore";
+import {getDownloadURL, ref, uploadBytes} from "firebase/storage";
+import {useEffect, useState} from "react";
+
+// NOTE: (Dennis) For some reason, the enum values are warning for not being used
+export enum ProcessingState {
+  // eslint-disable-next-line no-unused-vars
+  Uploading,
+  // eslint-disable-next-line no-unused-vars
+  Processing,
+  // eslint-disable-next-line no-unused-vars
+  Generating,
+  // eslint-disable-next-line no-unused-vars
+  Finalizing,
+  // eslint-disable-next-line no-unused-vars
+  Ready,
+}
+
+export function useGenkitArticle() {
+  const [error, setError] = useState<UserNotification | null>(null);
+  const [status, setStatus] = useState<ProcessingState>(ProcessingState.Ready);
+
+  useEffect(() => {
+    console.log("[useGenkitArticle] " + status);
+  }, [status]);
+
+  async function createFromRecording(file: File): Promise<string | null> {
+    const collRef = collection(firestore, "tanam-documents");
+    const docRef = doc(collRef);
+    const docId = docRef.id;
+
+    setStatus(ProcessingState.Uploading);
+    const fileExt = file.name.substring(file.name.lastIndexOf("."));
+    const fileName = Date.now().toString();
+    const filePath = `tanam-documents/${docId}/GenAI/${fileName}${fileExt}`;
```

Comment:
Nice process to store the input of voice recording to storage 👍

Reply:
Thanks. I was thinking of a future scenario where we keep all sources that generated the article so that we can "re-run" the generation if necessary :-)